src/app/Fake.Core.Process/CmdLineParsing.fs

```diff
@@ -128,28 +128,72 @@ module Args =
     /// Read a windows command line string into its arguments
     let fromWindowsCommandLine cmd = CmdLineParsing.windowsCommandLineToArgv cmd
 
+open BlackFox.CommandLine
+
 /// Represents a list of arguments
 type Arguments = 
-    internal { Args : string array }
-    static member Empty = { Args = [||] }
+    internal { Args : CmdLine }
+    static member Empty = { Args = CmdLine.empty }
     /// See https://msdn.microsoft.com/en-us/library/17w5ykft.aspx
     static member OfWindowsCommandLine cmd =
-        { Args = Args.fromWindowsCommandLine cmd }
+        { Args = Args.fromWindowsCommandLine cmd |> Array.toList |> CmdLine.fromList }
 
     /// This is the reverse of https://msdn.microsoft.com/en-us/library/17w5ykft.aspx
-    member x.ToWindowsCommandLine = Args.toWindowsCommandLine x.Args
-    member x.ToLinuxShellCommandLine = Args.toLinuxShellCommandLine x.Args
+    member x.ToWindowsCommandLine = x.Args |> CmdLine.toString
+    member x.ToLinuxShellCommandLine = x.Args |> CmdLine.toList |> Args.toLinuxShellCommandLine
 
     /// Create a new arguments object from the given list of arguments
-    static member OfArgs (args:string seq) = { Args = args |> Seq.toArray }
+    static member OfArgs (args:string seq) = { Args = args |> CmdLine.fromSeq }
     /// Create a new arguments object from a given startinfo-conforming-escaped command line string.
     static member OfStartInfo cmd = Arguments.OfWindowsCommandLine cmd
     /// Create a new command line string which can be used in a ProcessStartInfo object.
-    member x.ToStartInfo = CmdLineParsing.toProcessStartInfo x.Args
+    member x.ToStartInfo = x.Args |> CmdLine.toString // |>  CmdLineParsing.toProcessStartInfo x.Args
 
 module Arguments =
-    let withPrefix s (a:Arguments) =
-        Arguments.OfArgs(Seq.append s a.Args)
+    let withPrefix (s:string seq) (a:Arguments) =
+        { Args = CmdLine.concat a.Args (CmdLine.fromSeq s) }
+        //Arguments.OfArgs(Seq.append s a.Args)
     let append s (a:Arguments) =
-        Arguments.OfArgs(Seq.append a.Args s)
+        { Args = a.Args |> CmdLine.appendSeq s }
+        //Arguments.OfArgs(Seq.append a.Args s)
+
+/// Forward API from https://github.com/vbfox/FoxSharp/tree/master/src/BlackFox.CommandLine
```

Comment:
I think I understand why but it mean that the API in FAKE might diverge at some point.

Also you're losing both the "." style API and doc comments

Reply:
If the APIs diverge something most likely breaks?

Reply:
Yes I was not speaking about breakage, mostly playing catch-up with whatever is added on one side needing an equivalent PR on the other